dot-plot/script.js

```diff
@@ -131,13 +128,47 @@ function drawGraphic() {
       return x(d.max)
     })
     .attr('cy', function(d) {
-      return y(d.name)
+      return Math.abs(x(d.max)-x(d.min))<2?y(d.name)+3:y(d.name)
     })
 
+    //create data labels if switched on in config file
+    if(config.essential.dataLabels.show==true){
+      svg.selectAll('text.dataLabels')
+      .data(graphic_data)
+      .join('text')
+      .attr('class','dataLabels')
+      .attr('x',(d) => x(d.min)+12)
+      .attr('y',(d)=> y(d.name)+5)
+      .text((d)=>d3.format(config.essential.dataLabels.numberFormat)(d.min)+"%")
+      .attr('fill', config.essential.colour_palette[0])
+      .style("font-size", "14px")
+      .style("font-weight", 600)
+
+      svg.selectAll('text.dataLabels1')
+      .data(graphic_data)
+      .join('text')
+      .attr('class','dataLabels1')
+      .text((d)=>d3.format(config.essential.dataLabels.numberFormat)(d.max)+"%")
+      .attr('fill', config.essential.colour_palette[1])
+      .style("font-size", "14px")
+      .style("font-weight", 600)
+      .attr('y', function(d) {
+        return Math.abs(d.max -d.min)<0.03 && Math.abs(x(d.max)-x(d.min))<2?y(d.name)+5:y(d.name)+5
```

Comment:
could simplify this to just `return y(d.name)+5`, unless you want to leave the ternary operator in as an example.